subsys/net/ip/tcp2.c

```diff
@@ -450,16 +450,22 @@ static const char *tcp_conn_state(struct tcp *conn, struct net_pkt *pkt)
 	return buf;
 }
 
-static uint8_t *tcp_options_get(struct net_pkt *pkt, int tcp_options_len)
+static uint8_t *tcp_options_get(struct net_pkt *pkt, int tcp_options_len,
+				uint8_t *buf, size_t buf_len)
 {
-	static uint8_t options[40]; /* TCP header max options size is 40 */
 	struct net_pkt_cursor backup;
+	uint8_t *options = buf;
```

Comment:
optional: this would read a bit better if options is set on success instead of cleared on failure.

For example:

uint8_t *options = NULL;

...
if (ret == 0) {
  options = buf;
}

Reply:
Sure, why not.